### PR TITLE
Administration - QGIS project list: add information from qgis-project-validator

### PR DESCRIPTION
**Pull request by mdouchin**
This PR uses the output files of the proprietary `qgis-project-validator` tool to add more information to the administration panel menu "QGIS projects":

* **Invalid layers** count and list of names
* **Memory used** to load the project (Mo)
* **Loading time** of the project (seconds)
* **QGIS Log file** written when loading the project

It searches for specific files `.my_project.qgs.json` and `.my_project.qgs.log` stored in a dedicated folder accessible by Lizmap Web Client. If at least one valid JSON file is found, the QGIS projects table shows the new columns.

A new right **sidebar** has also been added to shows the project informations with the project image.

Funded by: **Valabre** (Centre de gravité de la formation des métiers de la Sécurité Civile, de la Recherche, des Nouvelles Technologies et de la Prévention dans le domaine des risques naturels)

2022-04-26_124323_1359x947

### CODE REVIEW

File: lizmap/modules/lizmap/classes/lizmapServices.class.php [hunk 1]
```diff
@@ -345,6 +349,32 @@ public function getRootRepositories()
...
+    {
+        $qgisProjectsInspectionDataFolder = $this->qgisProjectsInspectionDataFolder;
+
+        if ($qgisProjectsInspectionDataFolder != '') {
+            // if path is relative, get full path
+            if ($qgisProjectsInspectionDataFolder[0] != '/' and $qgisProjectsInspectionDataFolder[1] != ':') {
+                $qgisProjectsInspectionDataFolder = realpath($this->varPath.$qgisProjectsInspectionDataFolder);
+            }
+            // add a trailing slash if needed
+            if (!preg_match('#/$#', $qgisProjectsInspectionDataFolder) && $qgisProjectsInspectionDataFolder !== false) {
```
Comment (laurentj) [inline lizmap/modules/lizmap/classes/lizmapServices.class.php:368]: It is faster I guess: `$qgisProjectsInspectionDataFolder = rtrim($qgisProjectsInspectionDataFolder, '/').'/';`

**Review by laurentj — CHANGES_REQUESTED** [review]
- since the beginning, we are using camelCase variable names. We should continue with this coding style. So I recommend to not use `$variable_name_like_this` in PHP, but `$variableNameLikeThis`
- I suggest an other name for `qgisProjectsInspectionDataFolder`: `qgisProjectsPrivateDataFolder`. If we want to reuse later this directory for data other than "inspection"...

File: lizmap/var/config/lizmapConfig.ini.php.dist [hunk 2]
```diff
@@ -39,5 +39,10 @@ cacheRootDirectory="/tmp/"
 
 ; path to find repositories
 ;rootRepositories="path"
+; path to find the QGIS projects inspection data
+; created by the qgis-project-validator tool
+; if not give, try to use the rootRepositories path
```
[the remark this answers is not in the document]
Comment (mdouchin) [reply lizmap/var/config/lizmapConfig.ini.php.dist]: Thanks, modified

**Review by Gustry — APPROVED** [review]
Already approved by giving you a cookie for this work today ;-)

**Review by laurentj — APPROVED** [review]
lgtm

**Comment by rldhont** [comment]
@mdouchin changelog ?

**Comment by Gustry** [comment]
> @mdouchin changelog ?

I think it's already done from his previous PR : https://github.com/3liz/lizmap-web-client/blob/master/CHANGELOG-3.6.md#added

**Comment by rldhont** [comment]
> > @mdouchin changelog ?
> 
> I think it's already done from his previous PR : https://github.com/3liz/lizmap-web-client/blob/master/CHANGELOG-3.6.md#added

OK

**Comment by mdouchin** [comment]
Changelog for 3.6 improved. I merge !

**Pull request merged**